carma_novatel_driver_wrapper/src/carma_novatel_driver_wrapper.cpp

```diff
@@ -0,0 +1,207 @@
+        fix_msg.err_track = 2.0 * msg->azimuth_stdev;
+        fix_msg.err_pitch = 2.0 * msg->pitch_stdev;
+        fix_msg.err_roll = 2.0 * msg->roll_stdev;
+
+        // Convert Velocity
+        std::vector<double> components, variances;
+        components.push_back(msg->north_velocity);
+        variances.push_back(2.0 * msg->north_velocity_stdev);
+
+        components.push_back(msg->east_velocity);
+        variances.push_back(2.0 * msg->east_velocity_stdev);
+
+        // std::tuple<double, double> speed_tuple = uncertainty_tools::computeVectorMagnitudeAndUncertainty(components, variances);
+        //fix_msg.speed = std::get<0>(speed_tuple);
+        fix_msg.climb = msg->up_velocity;
+
+        fix_msg.err_climb = 2.0 * msg->up_velocity_stdev;
+        //fix_msg.err_speed = std::get<1>(speed_tuple);
+
+        fix_fused_pub_->publish(fix_msg);
+
+    }
+
+    void CarmaNovatelDriverWrapper::imu_callback(const sensor_msgs::msg::Imu::UniquePtr msg)
+    {
+        last_gnss_msg_ = this->now();
+    }
+
+    void CarmaNovatelDriverWrapper::status_timer_callback(const rclcpp::TimerCallbackType& event)
+    {
+        rclcpp::Time now = this->now();
+        switch(status_.status)
+        {
+            case carma_driver_msgs::msg::DriverStatus::OFF :
+                if(last_gnss_msg_ != rclcpp::Time(0) && last_imu_msg_ != rclcpp::Time(0)){
+                    status_.status = carma_driver_msgs::msg::DriverStatus::OPERATIONAL;
+                }
+                break;
+            case carma_driver_msgs::msg::DriverStatus::OPERATIONAL :
+                if(now - last_gnss_msg_ > rclcpp::Duration(config_.gnss_timeout) || now - last_imu_msg_ > rclcpp::Duration(config_.imu_timeout)){
+                    status_.status = carma_driver_msgs::msg::DriverStatus::FAULT;
+                }
+                break;
+            case carma_driver_msgs::msg::DriverStatus::DEGRADED :
+                if(now - last_gnss_msg_ > rclcpp::Duration(config_.gnss_timeout) || now - last_imu_msg_ > rclcpp::Duration(config_.imu_timeout)){
+                    status_.status = carma_driver_msgs::msg::DriverStatus::FAULT;
+                }
+                // TODO add logic for checking satellite coverage and use that for degraded status
+                break;
+            case carma_driver_msgs::msg::DriverStatus::FAULT :
+                if(now - last_gnss_msg_ < rclcpp::Duration(config_.gnss_timeout) && now - last_imu_msg_ < rclcpp::Duration(config_.imu_timeout)){
+                    status_.status = carma_driver_msgs::msg::DriverStatus::OPERATIONAL;
+                }
+                break;
+            default:
+            throw std::invalid_argument("Unknown status type: " + std::to_string(status_.status));
+        }
+
+        status_pub_->publish(status_);
+    }
+    
+    carma_ros2_utils::CallbackReturn CarmaNovatelDriverWrapper::on_system_alert(const carma_msgs::msg::SystemAlert::UniquePtr msg)
```

Comment:
Not needed looks like a copy paste from subsystem controller please only copy over code you understand the usage of.